main.go

```diff
@@ -0,0 +1,422 @@
+	err = dbInstance.getDatabaseInfo(RDSClient)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to obtain DB instance (%s) information", dbInstance.DBInstanceIdentifier)
+	}
+
+	if dbInstance.getSetDBInstanceClass() {
+		log.Infof("Current DB instance class (%s)", DBInstanceClasses[dbInstance.SizeIndex])
+	} else {
+		return errors.Wrapf(err, "Existing DB instance class not in the supported list")
+	}
+
+	newClass, err := dbInstance.getNewClassType()
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get DB instance (%s) new class type", dbInstance.DBInstanceIdentifier)
+	}
+
+	if !dbInstance.IsClusterWriter {
+		log.Infof("DB instance (%s) is a reader with instance class (%s). Calling class upgrade", dbInstance.DBInstanceIdentifier, dbInstance.DBInstanceClass)
+
+		err := dbInstance.changeDatabaseClass(RDSClient, newClass)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to change DB Instance (%s) class", dbInstance.DBInstanceIdentifier)
+		}
+	} else {
+		log.Infof("DB instance (%s) is a writer with instance class (%s). Getting first available reader", dbInstance.DBInstanceIdentifier, dbInstance.DBInstanceClass)
+		var dbInstanceReader DBInstance
+		clusterMembers, err := dbInstance.getDBClusterMembers(RDSClient)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to get DB cluster members")
+		}
+		for _, member := range clusterMembers {
+			if strings.Contains(*member.DBInstanceIdentifier, "rds-db-instance-multitenant") {
+				if *member.DBInstanceIdentifier != dbInstance.DBInstanceIdentifier {
+					dbInstanceReader.DBInstanceIdentifier = *member.DBInstanceIdentifier
+				}
+			}
+		}
+		log.Infof("DB instance (%s) was selected for vertical scaling. Getting database information", dbInstanceReader.DBInstanceIdentifier)
+		err = dbInstanceReader.getDatabaseInfo(RDSClient)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to obtain DB instance (%s) information", dbInstanceReader.DBInstanceIdentifier)
+		}
+
+		if dbInstanceReader.getSetDBInstanceClass() {
+			log.Infof("Current DB instance class (%s)", DBInstanceClasses[dbInstanceReader.SizeIndex])
+		} else {
+			return errors.Wrapf(err, "Existing DB instance class not in the supported list")
+		}
+
+		if (dbInstance.SizeIndex + 1) > dbInstanceReader.SizeIndex {
+			err = dbInstanceReader.changeDatabaseClass(RDSClient, newClass)
+			if err != nil {
+				return errors.Wrapf(err, "Failed to change DB instance (%s) class", dbInstanceReader.DBInstanceIdentifier)
+			}
+		}
+
+		log.Infof("Initiating DB instance (%s) failover", dbInstanceReader.DBInstanceIdentifier)
+		err = dbInstanceReader.databaseFailover(RDSClient)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to failover DB instance (%s)", dbInstanceReader.DBInstanceIdentifier)
+		}
+	}
+	log.Info("Vertical scaling was successfully handled, deleting SQS message")
+
+	err = deleteSQSMessage(SQSClient, message)
+	if err != nil {
+		return errors.Wrapf(err, "failed tο delete SQS message")
+	}
```

Comment:
Can you describe what would happen if something failed before here? Would we want to run this part even if something failed? I am not sure what leaving the SQS message around would mean.

Reply:
@gabrieljackson Considering that all previous actions are error handled, if the run reaches this part it means that everything should be ok and the message can now be deleted. Do you have something specifically in mind when saying something failed?

Reply:
I was more wondering if this part should be run even when there was an error, or if we want to leave the message around. It sounds like we don't want to delete the SQS message if anything errors before here though so probably fine to leave as is.